make/autoconf/flags-cflags.m4

```diff
@@ -719,6 +719,10 @@ AC_DEFUN([FLAGS_SETUP_CFLAGS_CPU_DEP],
         # for all archs except arm and ppc, prevent gcc to omit frame pointer
         $1_CFLAGS_CPU_JDK="${$1_CFLAGS_CPU_JDK} -fno-omit-frame-pointer"
       fi
+    elif test "x$FLAGS_OS" = xmacosx; then
+      if test "x$FLAGS_CPU_ARCH" = xaarch64; then
+        $1_CFLAGS_CPU="${$1_CFLAGS_CPU} -arch arm64"
+      fi
```

Comment:
This and other additions of -arch arm64 are not needed.
If one cross-compiling then these options could be added via extra-c/ld/cxx-flags configure arguments.
But if one compiles directly on macos-aarch64 machine then these options are useless as it will compile for arm64 by default there